src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/UWP.cs

```diff
@@ -565,9 +570,26 @@ internal string LogoPathFromUri(string uri)
                     }
                     else
                     {
-                        ProgramLogger.LogException($"|UWP|LogoPathFromUri|{Package.Location}" +
-                                                    $"|{UserModelId} can't find logo uri for {uri} in package location: {Package.Location}", new FileNotFoundException());
-                        return string.Empty;
+                        var targetSizes = new List<int>{ 16, 24, 30, 36, 40, 44, 60, 72, 96, 256 };
+
+                        foreach (var factor in targetSizes)
+                        {
+                            paths.Add($"{prefix}.targetsize-{factor}{extension}");
+                            paths.Add($"{prefix}.targetsize-{factor}_{theme}{extension}");
+                            paths.Add($"{prefix}.{theme}_targetsize-{factor}{extension}");
+                        }
+
+                        var selectedIconPath = paths.FirstOrDefault(File.Exists);

```

Comment:
Does this mean the smallest size would return first?

Reply:
Yep, and reversing the logic is equally as bad. You have to wire in an ideal size, look for that value, and if non-existent, look higher for scale down.

Reply:
Thanks for pointing that out @crutkas. I need to reverse the array. @riverar, the targetsize represents the pixels and it would be ideal to get the one with the largest target size.

Target size - 16 pixels
![image](https://user-images.githubusercontent.com/28739210/83597534-688d9d00-a51c-11ea-900c-2abdd4d54a37.png)

Target size - 256 pixels
![image](https://user-images.githubusercontent.com/28739210/83597550-73483200-a51c-11ea-8823-05d945dc81aa.png)

Also note, this case is hit only for **snip & sketch**. Most UWP application Images, calculator included, have Images with scale instead.

Reply:
That's not actually true. The ideal icon would be one closest to the size you're trying to display. The farther you deviate from the size, the more you're at the mercy of the image scaling algorithms in play here. Here's a visual example:

![image](https://user-images.githubusercontent.com/475132/83598966-ed2dea80-a51f-11ea-918d-bffaaa981c2e.png)

Observe when the 256px icon is resized down to 32px, it loses its curves, misrepresents some of the button highlights, and loses LCD shadow quality.

Reply:
Thank you for the visual @riverar. I shall modify the code to find the image closest to the size of the icon.